pkgs/stdenv/adapters.nix

```diff
@@ -144,7 +144,7 @@ rec {
     (mkDerivationSuper args).overrideAttrs (prevAttrs:
       if prevAttrs ? env.NIX_CFLAGS_LINK then {
         env = prevAttrs.env // {
-          NIX_CFLAGS_LINK = toString args.env.NIX_CFLAGS_LINK
+          NIX_CFLAGS_LINK = toString (args.env.NIX_CFLAGS_LINK or "")
```

Comment:
Why is this commit in this PR? If it's fixing an issue introduced by a commit in this PR, it should be either squashed into that commit, or ordered before it, so we never have a broken middle state.